### PR TITLE
Readline and README

### PR DESCRIPTION
**Pull request by Dymirt**
This commit is adding deadline library to the project and Update README file with project requirements.
Features:
- conditionally changes readline path in makefile based on OS

### CODE REVIEW

**Comment by Dymirt** [comment]
Hi! Waiting for review.

**Comment by Dymirt** [comment]
There is no need to copy all data from env using malloc.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# This project is about creating a simple shell.
+
+## TO DO
```
Comment (hipoliciak) [inline README.md:3]: Great idea to add a to-do list :)

File: srcs/driver/readline_wraper.c [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+/* ************************************************************************** */
+
+#include "minishell.h"
+
+char	*ft_read_line(char *line)
+{
+	if (line)
+	{
+		free (line);
+		line = (char *) NULL;
```
Comment (hipoliciak) [inline srcs/driver/readline_wraper.c:20]: Why are you casting NULL to char* here? From what I checked, the C standard guarantees that NULL can be assigned to any pointer type without an explicit cast.

File: srcs/driver/shell.c [hunk 3]
```diff
@@ -41,8 +43,36 @@ t_shell	init_shell(char **env)
...
+	free(shell->pipe_groups);
+}
+
 int	run_shell(t_shell *shell)
 {
+	char	*line;
+
 	(void)shell;
+	signal(SIGINT, sigint_handler);
+	line = (char *) NULL;
```
Comment (hipoliciak) [inline srcs/driver/shell.c:63]: Same comment as above.

**Review by hipoliciak — APPROVED** [review]
Except those two comments about casting NULL looks good. I'll see how it works after merging.

File: README.md (hunk 1 above)
Comment (Dymirt) [reply README.md:3]: Thnx

**Comment by Dymirt** [comment]
Thank you for reviving PR. You right, casting NULL is redundant but not incorrect. Il fix it in next PR.

**Pull request merged**